tests/crashes/recursive-print-issue-93775.rs

```diff
@@ -0,0 +1,38 @@
+//! This test case is modified from <https://github.com/rust-lang/rust/issues/93775>.
+//! The type printing involves recursive calls that lead to stack overflow.
+//! If it no longer crashes, please increase the nested type levels
+//! unless you are fixing this issue.
+//@ known-bug: #93775
+
+#![recursion_limit = "2049"]
```

Comment:
If I change to `16384`, my M2 Mac mini still executes successfully, though it takes 3 minutes to complete. This looks like another recursion?